src/SFML/Window/Android/JniHelper.hpp

```diff
@@ -145,16 +145,16 @@ class JniInputDevice
 class JniInputDeviceClass
 {
 public:
-    [[nodiscard]] static std::optional<JniInputDeviceClass> findClass(JNIEnv* env);
+    [[nodiscard]] static std::optional<JniInputDeviceClass> findClass(JNIEnv& env);
 
     [[nodiscard]] std::optional<JniArray<jint>> getDeviceIds();
 
     [[nodiscard]] std::optional<JniInputDevice> getDevice(jint idx);
 
 private:
-    JniInputDeviceClass(JNIEnv* env, jclass inputDeviceClass, jmethodID getDeviceIdsMethod, jmethodID getDeviceMethod);
+    JniInputDeviceClass(JNIEnv& env, jclass inputDeviceClass, jmethodID getDeviceIdsMethod, jmethodID getDeviceMethod);
 
-    JNIEnv*   m_env;
+    JNIEnv&   m_env;
```

Comment:
I'm not asking you to change how data is stored in the class. If it's important that these classes be reassignable, we can keep using a pointer instead of a reference.